src/App.tsx

```diff
@@ -1,223 +1,214 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { FilterValues } from './types/FilterValues';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  postTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterValue, setFilterValue] = useState(FilterValues.ALL);
+  const [title, setTitle] = useState('');
+  const [isTodoAdded, setIsTodoAdded] = useState(false);
+  const [selectedTodosIds, setSelectedTodosIds] = useState<number[]>([]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodosAsync = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch {
+        setErrorMessage('Unable to load todos');
+      }
+    };
+
+    if (user) {
+      getTodosAsync(user.id);
     }
   }, []);
 
+  const hundleAddTodo = useCallback(async (inputTitle: string) => {
+    setTitle(inputTitle);
+    setIsTodoAdded(true);
+
+    try {
+      if (user) {
+        const newTodo = await postTodo(user.id, inputTitle);
+
+        setTodos(prevTodos => [...prevTodos, newTodo]);
```

Comment:
let's call method for receiving actual todos from server instead of adding new todo manually in state